src/managers/hub/NAVManager.sol

```diff
@@ -0,0 +1,249 @@
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Holding updates
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc INAVManager
+    function updateHoldingValue(PoolId poolId, ShareClassId scId, AssetId assetId) external {
+        hub.updateHoldingValue(poolId, scId, assetId);
+    }
+
+    /// @inheritdoc INAVManager
+    function updateHoldingValuation(PoolId poolId, ShareClassId scId, AssetId assetId, IValuation valuation)
+        external
+        onlyManager(poolId)
+    {
+        hub.updateHoldingValuation(poolId, scId, assetId, valuation);
+        hub.updateHoldingValue(poolId, scId, assetId);
+    }
+
+    /// @inheritdoc INAVManager
+    function closeGainLoss(PoolId poolId, uint16 centrifugeId) external onlyManager(poolId) {
+        require(initialized[poolId][centrifugeId], NotInitialized());
+
+        AccountId equityAccount_ = equityAccount(centrifugeId);
+        AccountId gainAccount_ = gainAccount(centrifugeId);
+        AccountId lossAccount_ = lossAccount(centrifugeId);
+
+        (bool gainIsPositive, uint128 gainValue) = accounting.accountValue(poolId, gainAccount_);
+        (bool lossIsPositive, uint128 lossValue) = accounting.accountValue(poolId, lossAccount_);
+
+        accounting.unlock(poolId);
+
+        // Because we're crediting the gain account for gains and debiting the loss account for losses (and loss is
+        // credit-normal), gain should never be negative, and loss should never be positive.
+        // Still, double-check here.
+        if (gainIsPositive && gainValue > 0) {
+            accounting.addDebit(gainAccount_, gainValue);
+            accounting.addCredit(equityAccount_, gainValue);
+        }
+
+        if (!lossIsPositive && lossValue > 0) {
+            accounting.addCredit(lossAccount_, lossValue);
+            accounting.addDebit(equityAccount_, lossValue);
+        }
+
+        accounting.lock();
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Calculations
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc INAVManager
+    function netAssetValue(PoolId poolId, uint16 centrifugeId) public view returns (uint128) {
+        (bool equityIsPositive, uint128 equity) = accounting.accountValue(poolId, equityAccount(centrifugeId));
+        (bool gainIsPositive, uint128 gain) = accounting.accountValue(poolId, gainAccount(centrifugeId));
+        (bool lossIsPositive, uint128 loss) = accounting.accountValue(poolId, lossAccount(centrifugeId));
+        (bool liabilityIsPositive, uint128 liability) = accounting.accountValue(poolId, liabilityAccount(centrifugeId));
+
+        require(
+            equityIsPositive && gainIsPositive && liabilityIsPositive && (!lossIsPositive || loss == 0), InvalidNAV()
```

Comment:
Should take into accoutn `gain == 0` as we do with `loss == 0` in this check?

I mean `.. && (gainIsPositive || gain == 0) && ..`

Reply:
Thanks for clarifying this!